libpromises/mod_common.c

```diff
@@ -492,6 +492,7 @@ const ConstraintSyntax CF_COMMON_BODIES[] =
     ConstraintSyntaxNewString("if", "", "Extended classes ANDed with context", SYNTAX_STATUS_NORMAL),
     ConstraintSyntaxNewString("unless", "", "Negated 'if'", SYNTAX_STATUS_NORMAL),
     ConstraintSyntaxNewStringList("meta", "", "User-data associated with policy, e.g. key=value strings", SYNTAX_STATUS_NORMAL),
+    ConstraintSyntaxNewString("promiser_attribute", "", "Specify the promiser as an attribute", SYNTAX_STATUS_NORMAL),
```

Comment:
I don't think it would work without being an attribute because of the way variable and function resolution works. You can take my PR and work from that, I guess... I just don't know all the places it could break. I'll fix it up as you suggested and you can see what you think.